lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,13 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if Regex.match?(~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/, info) do
+      Mix.raise """
+      Your #{lockfile} contains merge conflicts.
+               Run `git checkout HEAD -- #{lockfile}` first to get a clean lock.
```

Comment:
I was wondering what to do here, `** (Mix) ` is 9 chars, so I indented the same amount to get an output like this:

```sh
** (Mix) Your mix.lock contains merge conflicts.
         Run `git checkout HEAD -- mix.lock` first to get a clean lock.
```

What do you think?